frontend/server/src/Controllers/QualityNomination.php

```diff
@@ -835,6 +835,31 @@ public static function apiResolve(\OmegaUp\Request $r): array {
         return ['status' => 'ok'];
     }
 
+    /**
+     * Updates a QualityNominations given an id and contents
+     *
+     * @param \OmegaUp\Request $r         The request.
+     *
+     * @return array{status: string}
+     */
+    public static function apiUpdate(\Omegaup\Request $r): array {
+        \OmegaUp\Controllers\Controller::ensureNotInLockdown();
+
+        // Validate request
+        $r->ensureMainUserIdentity();
+        self::validateMemberOfReviewerGroup($r);
+
+        $qualityNominationId = $r->ensureInt('qualitynomination_id');
+        \OmegaUp\Validators::validateStringNonEmpty($r['contents'], 'contents');
+
+        \OmegaUp\DAO\QualityNominations::updateQualityNominations(
+            $qualityNominationId,
+            $r['contents']
+        );
```

Comment:
```suggestion
        $rawContents = $r->ensureString(
            'contents',
            fn (string $alias) => \OmegaUp\Validators::stringNonEmpty($alias),
        );
        
        // Nota: de preferencia hay que hacer refactor con L647 y extraer esto a una función que ambas funciones puedan llamar para no tener tanto código duplicado.
        /**
         * @var null|array{tags?: mixed, before_ac?: mixed, difficulty?: mixed, quality?: mixed, statements?: mixed, source?: mixed, reason?: mixed, original?: mixed} $contents
         */
        $contents = json_decode($rawContents, associative: true);
        if (!is_array($contents)) {
            throw new \OmegaUp\Exceptions\InvalidParameterException(
                'parameterInvalid',
                'contents'
            );
        }

        \OmegaUp\DAO\QualityNominations::updateQualityNominations(
            $qualityNominationId,
            $rawContents,
        );
```

aún hay muchas referencias a `$r['...']`, pero ese patrón se prefiere evitarlo en código nuevo. también ese códi